modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -880,12 +892,21 @@ private Map<String, String> _addAction(
 					objectRelationshipElementsParser.parse(
 						objectRelationship, properties.get(entry.getKey()));
 
-				if (!nestedObjectEntries.isEmpty()) {
+				if (!_isManyToOneObjectRelationship(
+						relatedObjectDefinition, objectRelationship,
+						objectDefinition)) {
+
 					disassociateRelatedModels(
 						objectDefinition, objectRelationship, primaryKey,
 						relatedObjectDefinition,
 						dtoConverterContext.getUserId());
 				}
+				else {
+					_disassociateRelatedModel(
```

Comment:
This is very confusing. I don't know the difference between the public method and the private method as boths are called the same way